deebot_client/messages/json/map.py

```diff
@@ -0,0 +1,96 @@
+"""Map set v2 messages."""
+from __future__ import annotations
+
+import json
+from typing import TYPE_CHECKING, Any
+
+from deebot_client.events.map import MapSetEvent, MapSetType, MapSubsetEvent
+from deebot_client.message import HandlingResult, HandlingState, MessageBodyDataDict
+from deebot_client.util import decompress_7z_base64_data
+
+if TYPE_CHECKING:
+    from deebot_client.event_bus import EventBus
+
+
+class OnMapSetV2(MessageBodyDataDict):
+    """On map set v2 message."""
+
+    name = "onMapSet_V2"
+
+    @classmethod
+    def _handle_body_data_dict(
+        cls, event_bus: EventBus, data: dict[str, Any]
+    ) -> HandlingResult:
+        """Handle message->body->data and notify the correct event subscribers.
+
+        :return: A message response
+        """
+        # check if type is know
+        if not MapSetType.has_value(data["type"]):
+            return HandlingResult.analyse()
+
+        # if subsets is not given, it was an event/atr handling (this is to be done)
+        if not data.get("subsets") and data.get("mid"):
+            # NOTE: here would be needed to call 'GetMapSetV2' again with 'mid' and 'type',
+            #       that on event will update the map set changes,
+            #       messages current cannot call commands again
+            return HandlingResult(
+                HandlingState.SUCCESS, {"mid": data["mid"], "type": data["type"]}
+            )
+
+        # subset is based64 7z compressed
+        subsets: list[list[str]] = json.loads(
```

Comment:
If subsets are returned only by the get command than this code should be part of the command class and not message